conformance/lang/expressions/trap_expr.balt

```diff
@@ -0,0 +1,369 @@
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a function call expression.
+Labels: array-type, error-constructor-expr, error-type, function-call-expr, function-defn, int, optional-type, return-stmt, trap-expr, union-type
+
+function foo() returns int {
+    int[] i = [];
+    return i[3];
+}
+
+function bar() {
+    panic error("msg");
+}
+
+function init() {
+    int|error t1 = trap foo();
+    io:println(t1); // @output error("{ballerina/lang.array}IndexOutOfRange",message="array index out of range: index: 3, size: 0")
+    error? t2 = trap bar();
+    io:println(t2); // @output error("msg")
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a type cast expression.
+Labels: error-type, function-call-expr, function-defn, int, return-stmt, string, trap-expr, type-cast-expr, union-type
+
+function foo() returns int|string {
+    return "str";
+}
+
+function init() {
+    int|error t1 = trap <int>foo();
+    io:println(t1); // @output error("{ballerina}TypeCastError",message="incompatible types: 'string' cannot be cast to 'int'")
+    string|error t2 = trap <string>foo();
+    io:println(t2); // @output str
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a unary logical expression.
+Labels: boolean, error-type, function-call-expr, function-defn, int, return-stmt, trap-expr, union-type
+
+function foo() returns int|boolean {
+    return 23;
+}
+
+function init() {
+    boolean|error t1 = trap !<boolean>foo();
+    io:println(t1); // @output error("{ballerina}TypeCastError",message="incompatible types: 'int' cannot be cast to 'boolean'")
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a multiplicative expression.
+Labels: boolean, error-type, function-call-expr, function-defn, int, multiplicative-expr, trap-expr, union-type
+
+function init() {
+    int zero = 0;
+    int|error t1 = trap 2 / zero;
+    io:println(t1); // @output error("{ballerina}DivisionByZero",message=" / by zero")
+    int|error t2 = trap int:MAX_VALUE * int:MAX_VALUE;
+    io:println(t2); // @output error("{ballerina}NumberOverflow",message="int range overflow")
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a trap expression.
+Labels: boolean, error-type, function-call-expr, function-defn, int, multiplicative-expr, trap-expr, union-type
+
+function init() {
+    int zero = 0;
+    int|error t1 = trap trap 2 / zero;
+    io:println(t1); // @output error("{ballerina}DivisionByZero",message=" / by zero")
+}
+
+Test-Case: output
+Description: Test trap expression when the expression is a checking expression.
+Labels: boolean, check, checkpanic, error-type, int, multiplicative-expr, trap-expr, union-type
+
+function init() {
+    int zero = 0;
+    int|error e1 = trap (check 2 / zero);
```

Comment:
I think this is a bug. `trap` captures a panic. `check` should return the error from the function. Can you recheck in the spec and create a spec issue if required or a lang issue?